Add documentation about approval / validation requirements for DAO proposals (#2599)
nucypher/nucypher: docs/source/architecture/dao.rst

.. image:: ../.static/img/dao_proposal.png
    :target: ../.static/img/dao_proposal.png

Clicking on ``Yes`` or ``No`` triggers an Ethereum transaction signalling each staker's opinion on the proposal. That

> staker's opinion is weighted by their stake i.e. if they have 30,000 NU staked, then voting ``Yes`` or ``No`` will have
> a weight of 30,000 NU tokens.

Reviewer: ```suggestion
staker's opinion is weighted by their stake i.e. if they have 30,000 NU staked, then that staker will have
a weight of 30,000 NU tokens.
```

Depending on whether a quorum is reached, proposals may be validated or rejected. A proposal can only be validated
if and only if:

> #. it receives > 60% support i.e. of all the tokens that voted, more than 60% are required to vote ``Yes``
> #. it receives > 25% approval i.e. more than 25% of all the staked tokens that could have possibly voted are required to vote ``Yes``

Reviewer: ```suggestion
#. it receives > 60% support, i.e., of all the tokens that participated, more than 60% are required to signal ``Yes``
#. it receives > 25% approval, i.e., more than 25% of all the staked tokens that could have possibly participated are required to signal ``Yes``
```
